packages/wallet-ui/src/components/ui/molecule/AmountInput/AmountInput.view.tsx

```diff
@@ -110,29 +103,29 @@ export const AmountInputView = ({
   };
 
   const handleMaxClick = () => {
-    if (inputRef.current && asset.usdPrice) {
+    if (value && asset.usdPrice) {
       const amountStr = ethers.utils
         .formatUnits(asset.amount, asset.decimals)
         .toString();
       const amountFloat = parseFloat(amountStr);
-      inputRef.current.value = usdMode
+      const value = usdMode
         ? getAmountPrice(asset, amountFloat, false)
         : amountStr;
       fetchTotalPrice();
       resizeInput();
-      triggerOnChange();
+      triggerOnChange(value);
     }
   };
 
   useEffect(() => {
-    if (inputRef.current?.value && inputRef.current?.value !== '.') {
-      inputRef.current.value = totalPrice;
-      fetchTotalPrice();
-      triggerOnChange();
-      resizeInput();
+    if (value !== undefined) {
+      setInputValue(value);
     }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [usdMode]);
+  }, [value]);
+
+  useEffect(() => {
+    fetchTotalPrice();
+  }, [fetchTotalPrice]);
```

Comment:
```suggestion
  }, [asset, inputValue, usdMode]);
```

remove the useCallback , but move it to useEffect for easier understand, that useeffect is only trigger when xxxx